hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java

```diff
@@ -155,7 +155,8 @@ public TransactionBody decodeTransferNfts(
         final Address[] from = call.get(1);
         final Address[] to = call.get(2);
         final long[] serialNo = call.get(3);
-        if (from.length != to.length || from.length != serialNo.length) {
+
+        if ((from.length != to.length || from.length != serialNo.length) && from.length != 0) {
```

Comment:
Why do we need this `from.length != 0` check? Shouldn't we throw an exception if the `from` array is empty and the `to` array isn't?

Reply:
This is in order to get the EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS status in a child record, as it would never reach the logic in the handler for it, if we throw the exception here.

In Mono, we have a child record with EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS for this case. 
Previously, before this check in modularization, we had the check in the handler, but since we throw an error here, we never dispatch the child and never receive the error.

Reply:
Additional note:
The problem is that the [validation check](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java#L123) we have would never be reached if we throw the exception here. Mentioning Mono behavior is just for comparison. Since the difference here is not only the status (if it were, we might address it if differential testing prompts us), we may consider fixing it.

Reply:
I still don't understand. 😞 Are you saying we have a EET that:
 1. Uses a `from.length > 0`; and, 
 2. Uses a `to.length == 0`; and,
 3. Expects a child record with status `EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS`?

Reply:
This PR is adding such a EET.

Reply:
> Additional note: This [validation check](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java#L123) would never be reached if we throw the exception here.

BTW, the comment at that line seems wrong for the check as given:

> `// Verify that one and only one of the two types of transfers (fungible or non-fungible) is present`